server/src/main/java/org/elasticsearch/index/fielddata/fieldcomparator/DoubleValuesScriptComparatorSource.java

```diff
@@ -0,0 +1,138 @@
+
+        final double dMissingValue = (Double) missingObject(missingValue, reversed);
+        // NOTE: it's important to pass null as a missing value in the constructor so that
+        // the comparator doesn't check docsWithField since we replace missing values in select()
+        return new DoubleComparator(numHits, null, null, reversed, false) {
+            @Override
+            public LeafFieldComparator getLeafComparator(LeafReaderContext context) throws IOException {
+                NumberSortScript leafScript = scriptFactory.newInstance(new DocValuesDocReader(searchLookup, context));
+                return new DoubleLeafComparator(context) {
+                    @Override
+                    protected NumericDocValues getNumericDocValues(LeafReaderContext context, String field) throws IOException {
+                        return getNumericDoubleDocValues(context, dMissingValue, () -> getValuesFromScript(leafScript))
+                            .getRawDoubleValues();
+                    }
+
+                    @Override
+                    public void setScorer(Scorable scorer) {
+                        leafScript.setScorer(scorer);
+                    }
+                };
+            }
+        };
+    }
+
+    @Override
+    public BucketedSort newBucketedSort(
+        BigArrays bigArrays,
+        SortOrder sortOrder,
+        DocValueFormat format,
+        int bucketSize,
+        BucketedSort.ExtraData extra
+    ) {
+        return new BucketedSort.ForDoubles(bigArrays, sortOrder, format, bucketSize, extra) {
+            private final double dMissingValue = (Double) missingObject(missingValue, sortOrder == SortOrder.DESC);
+
+            @Override
+            public Leaf forLeaf(LeafReaderContext leafReaderContext) throws IOException {
+                NumberSortScript leafScript = scriptFactory.newInstance(new DocValuesDocReader(searchLookup, leafReaderContext));
+                return new Leaf(leafReaderContext) {
+                    private final NumericDoubleValues docValues = getNumericDoubleDocValues(
+                        leafReaderContext,
+                        dMissingValue,
+                        () -> getValuesFromScript(leafScript)
+                    );
+                    private double docValue;
+
+                    // TODO this is weird, why don't we just call the script? Seems like this is how it was done prior...
+                    @Override
+                    protected boolean advanceExact(int doc) throws IOException {
+                        if (docValues.advanceExact(doc)) {
+                            docValue = docValues.doubleValue();
+                            return true;
+                        }
+                        return false;
+                    }
+
+                    @Override
+                    protected double docValue() {
+                        return docValue;
+                    }
```

Comment:
As the comment mentions, this is kind of strange.  I wonder why we don't just call the script, similar to where these functions are overridden above?  Maybe there's a good reason, and I'll need to dig into it further.